src/world.rs

```diff
@@ -390,28 +391,49 @@ impl<L> LazyInsert for Vec<L>
     }
 }
 
-trait LazyInsertInternal: Send + Sync {
-    fn insert(self: Box<Self>, world: &World);
+struct LazyRemovalMarker<F>(PhantomData<F>);
+
+impl<F> LazyRemovalMarker<F> {
+    pub fn new() -> Self {
+        LazyRemovalMarker(PhantomData)
+    }
+}
+
+struct LazyRemovalInfo<C> {
+    entity: Entity,
+    marker: LazyRemovalMarker<C>,
+}
+
+trait LazyUpdateInternal: Send + Sync {
+    fn update(self: Box<Self>, world: &World);
 }
 
-impl<L> LazyInsertInternal for L
+impl<L> LazyUpdateInternal for L
     where L: LazyInsert
 {
-    fn insert(self: Box<Self>, world: &World) {
+    fn update(self: Box<Self>, world: &World) {
         L::insert(*self, world);
     }
 }
 
-/// Lazy insertions can be used after creating a new
+impl<C> LazyUpdateInternal for LazyRemovalInfo<C>
+    where C: Component + Send + Sync
+{
+    fn update(self: Box<Self>, world: &World) {
+        world.write::<C>().remove(self.entity);
+    }
+}
+
+/// Lazy updates can be used after creating a new
```

Comment:
I'd say in general, they can be used for stuff that should better happen at the end because it needs to borrow many resources. Dealing with initialization of components could be shown as an example of that.